google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -855,8 +858,10 @@ def tables token: nil, max: nil
         #   bigquery = Google::Cloud::Bigquery.new
         #   dataset = bigquery.dataset "my_dataset"
         #
-        #   job = dataset.query_job "SELECT name FROM my_table WHERE id = @id",
-        #                           params: { id: 1 }
+        #   job = dataset.query_job(
+        #     "SELECT name FROM my_table WHERE id = @id") do |query|
+        #     query.params = { id: 1 }
+        #   end
```

Comment:
Are you sure you want to change all of these examples to use the block updater? This is purely opinion, of course, but it seems to me that the previous approach with keyword args is much easier to read and understand.

Reply:
I thought we are moving to the block approach for all new properties, per https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1963#discussion_r167895215

If that is the case, I think we should be consistent for all properties.

Reply:
These [two `create_dataset` examples](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-bigquery/lib/google/cloud/bigquery/project.rb#L636-L653) show how keyword args can be used for "commonly-used" properties, and the block updater for "long-tail" properties. My understanding has always been that this hybrid usage is appropriate, although I don't know that I ever talked about it with anyone. @blowmage, what's your take?

Reply:
I think that the main options make sense to use as named arguments, but the less common ones may make more sense on the yielded object.

Reply:
In that case, I actually think `params` is a less commonly used argument. I could show some additional examples with the `write`, `legacy_sql`, and `table` options, which I think are going to be more common.

Reply:
Sounds great!

Reply:
My impression is that `params` is a pretty common argument, since it pairs directly with the required `query` argument. I think a less common argument would be `large_results` or `maximum_billing_tier`.

Reply:
Done in https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2003/commits/57309528c475abd94af9cc0ec97e14e000106472